apps/src/templates/instructions/TeacherFeedback.jsx

```diff
@@ -66,36 +137,66 @@ class TeacherFeedback extends Component {
   }
 
   componentDidMount = () => {
-    $.ajax({
-      url: `/api/v1/teacher_feedbacks/get_feedback_from_teacher?student_id=${
-        this.state.studentId
-      }&level_id=${this.props.serverLevelId}&teacher_id=${this.props.teacher}`,
-      method: 'GET',
-      contentType: 'application/json;charset=UTF-8'
-    })
-      .done((data, textStatus, request) => {
+    if (this.props.viewAs === ViewType.Student) {
+      $.ajax({
+        url:
+          '/api/v1/teacher_feedbacks/get_feedbacks?student_id=' +
+          this.props.user +
+          '&level_id=' +
+          this.props.serverLevelId,
+        method: 'GET',
+        contentType: 'application/json;charset=UTF-8'
+      }).done(data => {
         this.setState({
-          latestFeedback: request.status === 204 ? [] : [data],
-          token: request.getResponseHeader('csrf-token'),
-          comment: request.status === 204 ? '' : data.comment
+          latestFeedback: data,
+          comment: data[0].comment,
+          performance: data[0].performance
         });
-      })
-      .fail((jqXhr, status) => {
-        this.setState({errorState: ErrorType.Load});
       });
+    } else if (!this.props.disabledMode) {
+      $.ajax({
+        url: `/api/v1/teacher_feedbacks/get_feedback_from_teacher?student_id=${
```

Comment:
we can clean up this URL a little as well by pulling the props/state we need beforehand:

```
const {studentId} = this.state;
const {serverLevelId, teacher} = this.props;
`/api/v1/teacher_feedbacks/get_feedback_from_teacher?student_id=${studentId}&level_id=${serverLevelId}&teacher_id=${teacher}`
```